### ISSUE TITLE
color-contrast-matches isClipped function catches too much

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3037]
On https://www.drupal.nl/, none of the anchor elements in the top-nav are flagged by color-contrast (even though they should all fail). Turns out the parent `nav` element is marked as hidden by our `is-visible` function in the color-contrast-matcher because it uses a CSS clip but doesn't do it to try to hide the element from screen readers.

```css
.js .nav-collapse {
    clip: rect(0 0 0 0);
    max-height: none;
    position: relative;
    display: block;
    overflow: hidden;
    zoom: 1;
}
```

**Comment by clottman** [comment]
Clip only applies to absolutely positioned elements (`position: absolute` or `position: fixed`), and in this case the nav element is relatively positioned, so the clip attribute isn't actually doing anything. Note, clip-path can apply to any element regardless of positioning. 

So, I propose that our is-visible function, when checking isClipped, should also check positioning when an element `isClipped` because of the `clip` property.

**Comment by padmavemulapati** [comment]
Clipped elements which are styled as `position:absolute` and `position:fixed` not visible
Validated with the latest develop-branch code base
![image](https://user-images.githubusercontent.com/47409579/128515846-849ab8e0-f280-422d-abfa-1d4efe687d6b.png)

**Comment by straker** [comment]
No docs
